test/filterResults.test.ts

```diff
@@ -0,0 +1,79 @@
+import chai from 'chai'
+import chaiAsPromised from 'chai-as-promised'
+import chaiString from 'chai-string'
+import fs from 'fs/promises'
+import os from 'os'
+import path from 'path'
+import spawn from 'spawn-please'
+import { FilterResultsFunction } from '../src/types/FilterResultsFunction'
+import stubNpmView from './helpers/stubNpmView'
+
+chai.should()
+chai.use(chaiAsPromised)
+chai.use(chaiString)
+
+const bin = path.join(__dirname, '../build/src/bin/cli.js')
+
+/**
+ * Sets up and tears down the temporary directories required to run each test
+ */
+async function filterResultsTestScaffold(
+  dependencies: Record<string, string>,
+  filterResultsFn: FilterResultsFunction,
+  expectedOutput: string,
+  notExpectedOutput: string,
+): Promise<void> {
+  const stub = stubNpmView(
+    {
+      'ncu-test-v2': '3.0.0',
+      'ncu-test-tag': '2.1.0',
+      'ncu-test-return-version': '1.2.0',
+    },
+    { spawn: true },
+  )
+
+  // use dynamic import for ESM module
+  const { default: stripAnsi } = await import('strip-ansi')
+  const tempDir = await fs.mkdtemp(path.join(os.tmpdir(), 'npm-check-updates-'))
+  const pkgFile = path.join(tempDir, 'package.json')
+  await fs.writeFile(
+    pkgFile,
+    JSON.stringify({
+      dependencies,
+    }),
+    'utf-8',
+  )
+  const configFile = path.join(tempDir, '.ncurc.js')
+  await fs.writeFile(configFile, `module.exports = { filterResults: ${filterResultsFn.toString()} }`, 'utf-8')
+
+  try {
+    const stdout = await spawn('node', [bin, '--configFilePath', tempDir], {
```

Comment:
This is fine to run npm-check-updates in a child process, but note that this is a bit more verbose and less efficient than importing npm-check-updates as a module.

The only tests that run npm-check-updates as a child process are those that need to test functionality in the cli that can't be tested when importing it as a module.

Reply:
I think I got what you described here, but please check this part ;)